### ISSUE TITLE
industrial_robot_simulator:: trajectory_callback not correctly implemented when not using joint_groups

### ISSUE THREAD

**Issue by mmoerdijk** [ros-industrial-attic/industrial_experimental#7]
As far as I understand the industrial_experimental/industrial_robot_simulator should be a drop in replacement for the normal industrial_robot_simulator. On running it that way I got two errors.

First one says that `motion_ctrl` is not a member of `IndustrialRobotSimulatorNode` on line 627: where a call is made to `self.motion_ctrl.stop()`. I changed this to `motion_ctrl.stop()` (what I figured it should be) . Is this correct? Could it also be that line 623 and 627 are incorrectly indented? 

I got a error on line 631, where the following function call is made `point = self._to_controller_order(msg_in.joint_names,point)` saying that the function requires 4 instead of 2 arguments.  Looking at the definition `def _to_controller_order(self, ctrl_keys, traj_keys, traj_pt):` shows that the ctrl_keys and traj_keys arguments are missing. My setup does not require joint reordering so I removed the call to this function, this resulted in an working simulator.

**Comment by shaun-edwards** [comment]
@mmoerdijk, thanks for reporting the issue.

> First one says that motion_ctrl is not a member of IndustrialRobotSimulatorNode on line 627: where a call is made to self.motion_ctrl.stop(). I changed this to motion_ctrl.stop() (what I figured it should be) . Is this correct? Could it also be that line 623 and 627 are incorrectly indented?

From line [618](https://github.com/ros-industrial/industrial_experimental/blob/indigo-devel/IRC_v2/industrial_robot_simulator/industrial_robot_simulator#L618), it appears `motion_ctrl` is a local variable, so I believe you are correct.  623 and [627](https://github.com/ros-industrial/industrial_experimental/blob/indigo-devel/IRC_v2/industrial_robot_simulator/industrial_robot_simulator#L627) are indeed incorrectly indented.  It looks like line 627 should have been executing no matter what, which should have caused an error or at least always made the robot stop.  @JeremyZoss, could you take a look?  I'm not sure how this worked without @mmoerdijk's modifications.

> I got a error on line 631, where the following function call is made point = self._to_controller_order(msg_in.joint_names,point) saying that the function requires 4 instead of 2 arguments. Looking at the definition def _to_controller_order(self, ctrl_keys, traj_keys, traj_pt): shows that the ctrl_keys and traj_keys arguments are missing. My setup does not require joint reordering so I removed the call to this function, this resulted in an working simulator.

I'm glad you got this working.  We should try to track down the origin of these issues.  It's possible we didn't transfer the correct versions from our internal repos or some other sync issue.  @JeremyZoss, do you have any thoughts?

**Comment by shaun-edwards** [comment]
@JeremyZoss,  @phvass, friendly ping.

**Comment by JeremyZoss** [comment]
Sorry for the late response; I've been on vacation for 2 weeks.

This was indeed a sync error from our local repo to the `industrial_experimental` repo.  I tried to merge our local simulator updates with the public changes to `industrial_robot_simulator` since we forked the repo, and messed up the merge.  There was only minimal testing before uploading to `industrial_experimental`, and I missed this one.

The indention at 623 is correct as-is.  You get a different error message depending on the current robot state, but the currently-executing trajectory is always stopped when receiving a new trajectory.  The code here is almost the same as the [current simulator](https://github.com/ros-industrial/industrial_core/blob/indigo-devel/industrial_robot_simulator/industrial_robot_simulator#L375).

However, the lines from 625 to 632 were improperly merged.  I'll send a pull request with the correct code.
